subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/rules/IvyVariantComponentMetadataRulesIntegrationTest.groovy

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve.rules
+
+import org.gradle.integtests.fixtures.GradleMetadataResolveRunner
+import org.gradle.integtests.fixtures.RequiredFeature
+import org.gradle.integtests.fixtures.RequiredFeatures
+import org.gradle.integtests.resolve.AbstractModuleDependencyResolveTest
+
+@RequiredFeatures([
+    @RequiredFeature(feature = GradleMetadataResolveRunner.REPOSITORY_TYPE, value = "ivy"),
+    @RequiredFeature(feature = GradleMetadataResolveRunner.GRADLE_METADATA, value = "false"),
+])
+class IvyVariantComponentMetadataRulesIntegrationTest extends AbstractModuleDependencyResolveTest implements ComponentMetadataRulesSupport {
+
+    @Override
+    boolean isAddVariantDerivationRuleForIvy() { false }
+
+    def "opt-into variant aware dependency resolution if variant rules are added to ivy configuration"() {
+        given:
+        repository {
+            'org.test:projectA:1.0' {
+                variant 'customConf', [:] // in ivy, this is only a configuration
+            }
+        }
+
+        when:
+        buildFile << """
+            dependencies {
+                conf group: 'org.test', name: 'projectA', version: '1.0'
+                components {
+                    withModule('org.test:projectA') {
+                        withVariant('compile') { }
```

Comment:
What happens if I use `withDependencies()` here?
Still no attribute matching, but I can modify the dependencies declared in the source metadata, right?

Reply:
Yes